lib/system/dollars.nim

```diff
@@ -13,6 +13,7 @@ when not defined(nimPreviewSlimSystem):
     ## Outplace version of `addFloat`.
     result.addFloat(x)
 
+proc `$`*(x: pointer{lit}): string {.compileTime.} = "nil"
```

Comment:
`typeof(nil) | typeof(nil)` has been used for this

Reply:
duplicated?

Reply:
Sorry, I mean that is how the `nil` literal has been overloaded in the past, as an alternative to `pointer{lit}`.

Reply:
Yes, using two converts it to a typeclass which is the only way it works I believe

Reply:
duno which is better, my thought was user cant declare a const pointer, so `pointer{lit}` match the nil type and meaning.